packages/driver/cypress/integration/commands/actions/type_spec.js

```diff
@@ -177,6 +177,21 @@ describe('src/cy/commands/actions/type - #type', () => {
     .should('have.value', 'foobar')
   })
 
+  // https://github.com/cypress-io/cypress/issues/6125
+  it('works even if Event class is overridden', () => {
+    cy.visit('fixtures/issue-6125.html')
+    cy.get('#login_username')
+    .type('foobar')
+  })
+
+  // https://github.com/cypress-io/cypress/issues/5650
+  it('should trigger KeyboardEvent, not Event, for event listeners', () => {
+    cy.visit('fixtures/issue-5650.html')
+
+    cy.get('#test-input').type('A')
+    cy.get('#result').contains('isKeyboardEvent: true')
+  })
```

Comment:
I would suggest something like (to keep test smaller, faster):

```suggestion
  it('should trigger KeyboardEvent, not Event, for event listeners', () => {
    cy.$$('input:first').on('keydown', (e) => {
      if (e instanceOf e.currentTarget.ownerDocument.defaultView.KeyboardEvent) return
      throw new Error('event was not instanceOf KeyboardEvent')
    })
    cy.get('input:first').type('A')
  })
```

Reply:
I tried to reuse the test created in #5650. But I prefer your way.

But the code above has 2 problems:

1. `e` is not a native event. To access it, we need to use `e.originalEvent`.

> It's also important to note that the event object contains a property called `originalEvent`, which is the event object that the browser itself created. jQuery wraps this native event object with some useful methods and properties, but in some instances, you'll need to access the original event via `event.originalEvent` for instance. This is especially useful for touch events on mobile devices and tablets.

2. `done` function should be used. Otherwise, the test would pass without checking `keydown` event. Or fail later when Cypress is running other tests. 

I fixed these 2 problems and updated the test.